test/e2e/create_cluster_test.go

```diff
@@ -2170,6 +2170,88 @@ func TestOnCreateAPIUX(t *testing.T) {
 					},
 				},
 			},
+			{
+				name: "when AWS imageType and arch combinations are validated",
+				file: "nodepool-base.yaml",
+				validations: []struct {
+					name                   string
+					mutateInput            func(*hyperv1.NodePool)
+					expectedErrorSubstring string
+				}{
+					{
+						name: "should pass when imageType is not specified for arm64",
+						mutateInput: func(np *hyperv1.NodePool) {
+							np.Spec.Arch = hyperv1.ArchitectureARM64
+							np.Spec.Platform.Type = hyperv1.AWSPlatform
+							np.Spec.Platform.AWS = &hyperv1.AWSNodePoolPlatform{
+								InstanceType: "m6g.large",
+								// ImageType intentionally not set
+							}
+						},
+						expectedErrorSubstring: "",
+					},
+					{
+						name: "should pass when imageType is not specified for amd64",
+						mutateInput: func(np *hyperv1.NodePool) {
+							np.Spec.Arch = hyperv1.ArchitectureAMD64
+							np.Spec.Platform.Type = hyperv1.AWSPlatform
+							np.Spec.Platform.AWS = &hyperv1.AWSNodePoolPlatform{
+								InstanceType: "m6i.large",
+								// ImageType intentionally not set
+							}
+						},
+						expectedErrorSubstring: "",
+					},
+					{
+						name: "should fail when imageType is Windows with arm64 architecture",
+						mutateInput: func(np *hyperv1.NodePool) {
+							np.Spec.Arch = hyperv1.ArchitectureARM64
+							np.Spec.Platform.Type = hyperv1.AWSPlatform
+							np.Spec.Platform.AWS = &hyperv1.AWSNodePoolPlatform{
+								InstanceType: "m6g.large",
+								ImageType:    ptr.To(hyperv1.AWSNodePoolImageTypeWindows),
```

Comment:
```suggestion
								ImageType:    ptr.To(hyperv1.ImageTypeWindows),
```
see ImageType enum values [here](https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/nodepool_types.go#L41)